### ISSUE TITLE
[Request] PLEASE Make the app match/follow the material design specs (and other additional things)

### ISSUE THREAD

**Issue by jahirfiquitiva** [laurencedawson/reddit-sync-development#1576]
1. Slide for Reddit, has an option to tint subreddit name, or even the whole card, with the subreddit theme color, in the Frontpage... Which I think is really good to be able to recognize where the post comes from. ([Example 1](https://drive.google.com/file/d/0Bw52d3_ZiSb9OGNzUG1VZnFhd2s/view?usp=sharing) ... [Example 2](https://drive.google.com/file/d/0Bw52d3_ZiSb9M1lhME94RW45OHM/view?usp=sharing) ... Please note that with example 2, texts and icons are all white. I don't like that thing. Please check point 7 for my opinion about it. Thanks.)

2. Red color for NSFW posts is way too dark, and the text in it looks really bad, considering you ignored [guidelines](https://material.google.com/style/color.html#color-text-background-colors) and you are using black and grey for it.

3. Following point 2, use colors with opacity for texts. It's [adviced](http://prntscr.com/d0bgg8) in guidelines to do it. And [here](http://prntscr.com/d0bgw0) you have the proper values and opacities in case you haven't seen them. **This doesn't go just for texts but icons too.**

4. Update (and/or make sure) the colors for themes to match the [material design palette](https://material.google.com/style/color.html#color-color-palette) ... Some colors seem to be outdated.

5. In comment creation activity, there's no tinted status bar. Also, the accent color is teal, instead of theme based color or the custom color I chose in settings. And finally, its icons are tinted incorrectly, check the values in what I mentioned in point 3.

6. In comment creation activity too, if one discards the comment one was going to create, the comment is automatically copied to clipboard. Which is annoying because maybe one has something important in the clipboard. Please add an option to disable that behavior.

7. Keeping in mind points 1 and 3. I would suggest that, if option to tint cards background (in case you add it), is enabled... The app checks if the color is light or dark, and then use the proper color values (texts and icons) for light or dark themes, which can be found in the material design guidelines.

You can use this code to check if color is light:
```java
    public static boolean isLightColor(@ColorInt int color) {
        if (color == Color.BLACK) return 1.0;
        else if (color == Color.WHITE || color == Color.TRANSPARENT) return 0.0;
        return (1 - (0.299 * Color.red(color) + 0.587 * Color.green(color) + 0.114 * Color.blue(color)) / 255) < 0.4;
    }
```

##

I think besides these things, the app is really good, I really like it, but these things are important for me. They're minimal, but yet worth it changes, and being minimal, means easy to implement too. I really hope you consider it, and add them to the app for next update.

Thanks in advance, and keep up the good work.

**Comment by laurencedawson** [comment]
1. Has been added in the latest beta

**Comment by laurencedawson** [comment]
5. This is a limitation of the FragmentDialog Sync is using

**Comment by laurencedawson** [comment]
6. This has since been replaced with a drafts manager

**Comment by laurencedawson** [comment]
4. This one is incorrect. The colors shown in the picker are nearly all material colors:

public static int[] defaultColors = {
      0xFFF44336,  // Material orange
      0xFFE91E63,  // Material pink
      0xFF9C27B0,  // Material purple
      0xFF673AB7,  // Material deep purple
      0xFF3F51B5,  // Material indigo
      0xFF4484f6,  // Custom blue
      0xFF03A9F4,  // Material light blue
      0xFF00BCD4,  // Material cyan
      0xFF009688,  // Material teal
      0xFF4CAF50,  // Material green
      0xFF8BC34A,  // Material lime green
      0xFFCDDC39,  // Material lime
      0xFFFFEB3B,  // Material yellow
      0xFFFFC107,  // Material Amber
      0xFFFF9800,  // Material orange
      0xFFFF5722,  // Material deep orange

      0xFF697c8f,  // Custom
      0xFF263238,  // Custom
      0xFF222222,  // Very dark grey
      0xFF000000,  // Black
  };

**Comment by laurencedawson** [comment]
2. Updated the title color to white when in light mode

**Comment by jahirfiquitiva** [comment]
I just tried latest beta (Beta 4) and...

1. I can't seem to find this option anywhere.
2. Hasn't been fixed, and just changing the title color to white is just a work-around, instead of a real fix, which would be the use of proper text color values. (which I linked).
3. Same as 2.
4. Ok
5. I'm sure there's always a way to make it work, either creating a custom theme for that fragment, or changing the fragment implementation. So, saying is a limitation, is honestly an excuse to not do it.
6. Great
